packages/client/hmi-client/src/components/projects/NewProjectCard.vue

```diff
@@ -1,16 +1,49 @@
 <script setup lang="ts">
 import Button from '@/components/Button.vue';
 import IconAddFilled32 from '@carbon/icons-vue/es/add--filled/32';
+import Modal from '@/components/Modal.vue';
+import { ref } from 'vue';
+import { useRouter } from 'vue-router';
 
-function createNewProject() {}
+const isModalVisible = ref(false);
+const newProjectTitle = ref('New Project');
+const router = useRouter();
+
+function postProject(name: string) {
+	// Placeholder implementation - replace with api call
+	return {
+		id: '3',
+		name,
+		description: 'This is a new project'
+	};
+}
+
+function createNewProject() {
+	isModalVisible.value = false;
+	const { id } = postProject(newProjectTitle.value);
+	const routeLocation = router.resolve({ path: `/projects/${id}` });
+	window.open(routeLocation.href, '_blank');
+}
 </script>
 
 <template>
 	<div class="new-project-card">
-		<Button @onclick="createNewProject"
+		<Button @click="isModalVisible = true"
 			>Create New Project
 			<IconAddFilled32 />
 		</Button>
+		<Teleport to="body">
+			<modal :isVisible="isModalVisible">
+				<template #header>
+					<h3>Create a new project</h3>
+				</template>
+				<template #body>
+					<label for="input">Project title</label>
+					<input v-model="newProjectTitle" placeHolder="New Project" />
```

Comment:
* For some reason they are black on Firefox.
![Screen Shot 2022-11-16 at 14 56 23](https://user-images.githubusercontent.com/636801/202281286-b6e3d413-32b0-4ab7-a496-faad130acc6d.png)

* Could you add a `description` input as well

Reply:
Looks fine on Firefox on my system!